Split AnimationValue and friends out of GraphicsLayer

annevk opened the pull request:
#### 364c78497c2b79a80cdb153751b201b43d091df6
<pre>
Split AnimationValue and friends out of GraphicsLayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=304677">https://bugs.webkit.org/show_bug.cgi?id=304677</a>

Reviewed by Sam Weinig.

Address a couple of existing FIXMEs asking for these classes to be
moved out of GraphicsLayer to enable us to reduce unsafeness in a
subsequent change (so we can clearly attribute blame in case something
goes amiss).

We did take the liberty of making GraphicLayerAnimationValue&apos;s
m_timingFunction const as part of this change as that&apos;s trivial enough.

Canonical link: <a href="https://commits.webkit.org/304936@main">https://commits.webkit.org/304936@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8207d5e87bd4eb6c5b9fee98d50e734acc5b4fa1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136950 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9310 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48237 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144692 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89923 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3f972514-41b9-41c9-b416-535b7ff78905) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138822 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10013 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9156 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104722 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76415 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/46c6ebb3-31dc-4bc6-a409-0ed21d50085b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139895 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7333 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122708 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85560 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1aa3ad7d-5d36-4460-af04-772b62ec1c41) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/136298 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6970 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4676 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5280 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116305 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40885 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147446 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8993 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41454 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113077 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9011 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7557 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113407 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6897 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118984 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63227 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21107 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9041 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37038 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8762 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72607 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8982 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8834 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/platform/graphics/AnimationValue.h:
```diff
@@ -0,0 +1,66 @@
+// Base class for animation values (also used for transitions). Here to
+// represent values for properties being animated via the GraphicsLayer,
+// without pulling in style-related data from outside of the platform directory.
+class AnimationValue {
```
Given how tied to GraphicsLayer these are, how do you feel about renaming them to the slightly verbose, GraphicsLayerAnimationValue (and similar for derived types)?

annevk replied:
Fine with me. @smfr @graouts?

annevk replied:
Same for KeyframeValueList btw?

weinig replied:
Yeah.

annevk committed:
Split AnimationValue and friends out of GraphicsLayer
https://bugs.webkit.org/show_bug.cgi?id=304677

Reviewed by Sam Weinig.

Address a couple of existing FIXMEs asking for these classes to be
moved out of GraphicsLayer to enable us to reduce unsafeness in a
subsequent change (so we can clearly attribute blame in case something
goes amiss).

We did take the liberty of making GraphicLayerAnimationValue's
m_timingFunction const as part of this change as that's trivial enough.

Canonical link: https://commits.webkit.org/304936@main
